### PR TITLE
Use playwright instead of requests for browse

### PR DESCRIPTION
**Pull request by ryanmac**
Switch out the browse feature with Playwright. Many sites do not work with `requests`. Playwright imitates the browser better.

This also adds a new requirement for install:

```
pip install playwright
playwright install
```

This also includes:
- Hide `auto_gpt_workspace/*` in `.gitignore`
- Mac Python 3.11 support (removed dirtyjson in favor of json)
- Updated requirements.txt

### CODE REVIEW

File: .gitignore
Comment (nponeccop) [inline .gitignore:1]: Submit this change as a separate PR. PRs should be atomic, not mixing unrelated changes together.

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 beautifulsoup4
 colorama==0.4.6
-dirtyjson==1.0.
```
Comment (nponeccop) [inline requirements.txt]: Why? It is unrelated to the Playwrite replacements. Separate PR. Even if it was related, delete, not comment.

File: requirements.txt [hunk 2]
```diff
@@ -9,5 +9,5 @@ readability-lxml==0.8.1
 requests
 tiktoken==0.3.3
 docker
-# googlesearch-python
```
Comment (nponeccop) [inline requirements.txt]: Pinning down a dependency is unrelated. Put to a separate PR.

File: scripts/json_parser.py
Comment (nponeccop) [inline scripts/json_parser.py:1]: Using JSON instead of Dirty JSON should be either in a separate PR or documented in the description of this PR as for why it is important

**Review by nponeccop — DISMISSED** [review]
Split into atomic PRs.

**Comment by nponeccop** [comment]
@ryanmac Resolve the conflicts

**Comment by BillSchumacher** [comment]
Looks good for the most part if you do decide to update this move to commands/ and remove duplicate functions please.

**Review by BillSchumacher — APPROVED** [review]
